exchanges/graphcache/src/store/data.ts

```diff
@@ -161,11 +161,9 @@ export const clearDataState = () => {
     while (
       --i >= 0 &&
       data.refLock.has(data.optimisticOrder[i]) &&
-      data.commutativeKeys.has(data.optimisticOrder[i]) &&
-      !data.deferredKeys.has(data.optimisticOrder[i])
-    ) {
+      data.commutativeKeys.has(data.optimisticOrder[i])
```

Comment:
Here, as discussed, we realised that deferred layers should 100% be squasheable. We basially put deferred layers in the right place, write to them, and then ... we assume that they're never squasheable.

This means that for subscriptions (or other neverending result streams) this layer never goes away, and pending layers that then were on top can never go away either, and have the potential to infinitely be "in the way" of old deferred layers.